src/app_service/service/community/service.nim

```diff
@@ -482,6 +506,51 @@ QtObject:
       self.allCommunities[community.id] = community
       self.events.emit(SIGNAL_COMMUNITIES_UPDATE, CommunitiesArgs(communities: @[community]))
 
+      # tokenPermission was added
+      if community.tokenPermissions.len > prev_community.tokenPermissions.len:
+        for id, tokenPermission in community.tokenPermissions:
+          if not prev_community.tokenPermissions.hasKey(id):
+            self.allCommunities[community.id].tokenPermissions[id] = tokenPermission
+
+            if not community.admin:
```

Comment:
I don't think this condition makes sense. We'll have multiple admins in the future and you want to update the models when another admin added a permission.

IMO, the best way to handle it is to make sure that when we do `addPermission` (or whatever the name) below, we update `allCommunities`  communities accordingly, so the condition above `if community.tokenPermissions.len > prev_community.tokenPermissions.len:` doesn't even trigger for the current admin.

Reply:
Okay so... the reason I'm checking for `community.admin` here is because, we're already emitting the signal (as admin) in the service after creation/update/deletion, so that would result in the signal being emitted twice, and twice the amount of app notifications etc.

If however `community.admin` is no longer == "owner", then this indeed needs to change.

Another alternative would be to not emit the signal from the service in the first place, and purely rely on `handleCommunityUpdates` to trigger the signal here. That means however, that signal has to come in first, before we see a change in the UI (which might not be a problem, if we know that signal will happen synchronously via MessengerResponse, after the create/update permission via service was done)

> IMO, the best way to handle it is to make sure that when we do addPermission (or whatever the name) below, we update allCommunities communities accordingly, so the condition above if community.tokenPermissions.len > prev_community.tokenPermissions.len: doesn't even trigger for the current admin.

Not super sure I follow. We already update `allCommunities` right now but we still want to know if there was a change in the tokenPermissions to emit a signal (cause this could be a normal member node)

Reply:
> Another alternative would be to not emit the signal from the service in the first place, and purely rely on handleCommunityUpdates to trigger the signal here. That means however, that signal has to come in first, before we see a change in the UI (which might not be a problem, if we know that signal will happen synchronously via MessengerResponse, after the create/update permission via service was done)

This is not super reliable. There is often a delay between the signal comes back and when we do the action, so it creates a delay on the front-end.

Reply:
I believe the easiest path forward would be a distinction between `owner` and `admin`
Something that lets us express: "If you were the one who created/updated this permission, you don't need this signal"

Reply:
> Not super sure I follow. We already update allCommunities right now but we still want to know if there was a change in the tokenPermissions to emit a signal (cause this could be a normal member node)

Yeah don't worry, you already do the necessary update to `allCommunities`, so that means that ` if not prev_community.tokenPermissions.hasKey(id)` would not even trigger, as far as I can tell. Therefore, I don't think you need  `if not community.admin:`.